os/Makefile.unix

```diff
@@ -495,7 +475,56 @@ ifeq ($(CONFIG_RAW_BINARY),y)
 	$(Q) $(OBJCOPY) $(OBJCOPYARGS) -O binary $(BIN) $(BIN).bin
 endif
 
+define VERIFY_APP
+	$(Q) echo "Verify $(OUTBIN_DIR)/$1"
+	$(Q) if [ "`nm -u $(OUTBIN_DIR)/$1 | wc -l`" != "0" ]; then \
+		echo "Undefined Symbols in $(OUTBIN_DIR)/$1"; \
+		nm -u -l $(OUTBIN_DIR)/$1; \
+		rm $(OUTBIN_DIR)/$1; \
+		exit 1; \
+	fi
+endef
+
+ifeq ($(CONFIG_ELF_EXCLUDE_SYMBOLS),y)
+define STRIP_DBG
+	$(Q) $(OBJCOPY) --remove-section .comment $(OUTBIN_DIR)/$1
+	$(Q) $(STRIP) -g $(OUTBIN_DIR)/$1 -o $(OUTBIN_DIR)/$1
+endef
+endif
+
+ifeq ($(CONFIG_COMPRESSED_BINARY),y)
+define COMPRESS_BIN
+	$(Q) cp $(OUTBIN_DIR)/$1 $(OUTBIN_DIR)/$1.uncomp
+	$(Q) tools/compression/mkcompressimg $(CONFIG_COMPRESSION_BLOCK_SIZE) $(CONFIG_COMPRESSION_TYPE) $(OUTBIN_DIR)/$1 $(OUTBIN_DIR)/$1.comp
+	$(Q) mv $(OUTBIN_DIR)/$1.comp $(OUTBIN_DIR)/$1
+endef
+endif
+
+define INSTALL_APP
+	$(Q) echo "Install $(OUTBIN_DIR)/$1"
+	$(Q) cp $(OUTBIN_DIR)/$1 $(OUTBIN_DIR)/$1_dbg
+	$(Q) $(call STRIP_DBG,$1)
+	$(Q) $(call COMPRESS_BIN,$1)
+	$(Q) $(TOPDIR)/tools/mkbinheader.py $(OUTBIN_DIR)/$1 $2 $3 $1 $4 $5 $6 $7 $8
+	$(Q) $(TOPDIR)/tools/mkchecksum.py $(OUTBIN_DIR)/$1
+	$(Q) cp $(OUTBIN_DIR)/$1 $(OUTBIN_DIR)/user
+endef
+
 post:
+	$(Q) mkdir -p $(OUTBIN_DIR)/user
+ifneq ($(CONFIG_SUPPORT_COMMON_BINARY),y)
+	$(Q) $(call VERIFY_APP,"app1")
```

Comment:
Could you leave comment why we skip to verify the binary when `CONFIG_SUPPORT_COMMON_BINARY` is enabled?